py-requirements.txt

```diff
@@ -0,0 +1,43 @@
+ansible==9.3.0
+ansible-compat==4.1.11
+ansible-core==2.16.4
+ansible-generator==3.1.1
+ansible-lint==24.2.0
+argcomplete==3.2.3
+attrs==23.2.0
+black==24.2.0
+bracex==2.4
+certifi==2024.2.2
+cffi==1.16.0
+click==8.1.7
+cryptography==42.0.5
+filelock==3.13.1
+Jinja2==3.1.3
+jsonschema==4.21.1
+jsonschema-specifications==2023.12.1
+markdown-it-py==3.0.0
+MarkupSafe==2.1.5
+mdurl==0.1.2
+mypy-extensions==1.0.0
+packaging==24.0
+pathspec==0.12.1
+platformdirs==4.2.0
+pycparser==2.21
+Pygments==2.17.2
+PyYAML==6.0.1
+referencing==0.33.0
+resolvelib==1.0.1
+rich==13.7.1
+rpds-py==0.18.0
+ruamel.yaml==0.18.6
+ruamel.yaml.clib==0.2.8
+sentry-sdk==1.41.0
+subprocess-tee==0.4.1
+tomli==2.0.1
+tomlkit==0.12.4
+typing_extensions==4.10.0
+urllib3==2.2.1
+wcmatch==8.5.1
+xmltodict==0.13.0
+yamllint==1.35.1
+yq==3.2.3
```

Comment:
Not sure what others will think but is it necessary to have specific version for all of these dependencies. In case of future updates/patches we can remove the `==<version>` to use the latest version of each package. If it is necessary to use specific versions then I am happy to leave this as is.